### ISSUE TITLE
ad-Advice-select-window: Wrong type argument: window-live-p, nil

### ISSUE THREAD

**Issue by malb** [abo-abo/hydra#325]
I'm not sure against which package to open this issue, so apologies if I misfiled it.

I'm now seeing `ad-Advice-select-window: Wrong type argument: window-live-p, nil` when trying to display any hydra with https://github.com/yuya373/emacs-slack loaded. The backtrace is:

```
Debugger entered--Lisp error: (wrong-type-argument window-live-p nil)
  #<subr select-window>(nil norecord)
  ad-Advice-select-window(#<subr select-window> nil norecord)
  apply(ad-Advice-select-window #<subr select-window> (nil norecord))
  #f(advice-wrapper :around #<subr select-window> ad-Advice-select-window)(nil norecord)
  funcall(#f(advice-wrapper :around #<subr select-window> ad-Advice-select-window) nil norecord)
  slack-advice-select-window(#f(advice-wrapper :around #<subr select-window> ad-Advice-select-window) nil norecord)
  apply(slack-advice-select-window #f(advice-wrapper :around #<subr select-window> ad-Advice-select-window) (nil norecord))
  select-window(nil norecord)
  lv-message(#("Org:       o compact  O normal     t table\nMarkdown:  m compact  M normal\nOther:     p plain    c csv table  j json table x ocr\no, O, m, M, c, j, t, p, x, [q]: cancel." 11 12 (face hydra-face-blue) 22 23 (face hydra-face-blue) 35 36 (face hydra-face-blue) 54 55 (face hydra-face-blue) 65 66 (face hydra-face-blue) 85 86 (face hydra-face-blue) 96 97 (face hydra-face-blue) 109 110 (face hydra-face-blue) 122 123 (face hydra-face-blue) 128 129 (face hydra-face-blue) 131 132 (face hydra-face-blue) 134 135 (face hydra-face-blue) 137 138 (face hydra-face-blue) 140 141 (face hydra-face-blue) 143 144 (face hydra-face-blue) 146 147 (face hydra-face-blue) 149 150 (face hydra-face-blue) 152 153 (face hydra-face-blue) 156 157 (face hydra-face-blue)))
  hydra-show-hint((format #("Org:       %s compact  %s normal     %s table\nMarkdown:  %s compact  %s normal\nOther:     %s plain    %s csv table  %s json table %s ocr\no, O, m, M, c, j, t, p, x, [q]: cancel." 137 138 (face hydra-face-blue) 140 141 (face hydra-face-blue) 143 144 (face hydra-face-blue) 146 147 (face hydra-face-blue) 149 150 (face hydra-face-blue) 152 153 (face hydra-face-blue) 155 156 (face hydra-face-blue) 158 159 (face hydra-face-blue) 161 162 (face hydra-face-blue) 165 166 (face hydra-face-blue)) #("o" 0 1 (face hydra-face-blue)) #("O" 0 1 (face hydra-face-blue)) #("t" 0 1 (face hydra-face-blue)) #("m" 0 1 (face hydra-face-blue)) #("M" 0 1 (face hydra-face-blue)) #("p" 0 1 (face hydra-face-blue)) #("c" 0 1 (face hydra-face-blue)) #("j" 0 1 (face hydra-face-blue)) #("x" 0 1 (face hydra-face-blue))) malb/hydra-pdf-extract)
  malb/hydra-pdf-extract/body()
  funcall-interactively(malb/hydra-pdf-extract/body)
  call-interactively(malb/hydra-pdf-extract/body nil nil)
  command-execute(malb/hydra-pdf-extract/body)
```

Initially, it looks like an `emacs-slack` bug but the error is raised in the advice function it seems. Thus, perhaps hydra is calling `select-window` in a strange way? Feel free to simply close the issue if you think it's an `emacs-slack` bug and I'll open it against `emacs-slack`

**Comment by abo-abo** [comment]
This is not a hydra bug, I think. You can examine `(symbol-function 'malb/hydra-pdf-extract/body)` to investigate.

**Comment by malb** [comment]
Thanks for the quick reply, I'll try the emacs-slack folks.

**Comment by malb** [comment]
FYI: 

> According to the stack trace, nil is passed to select-window, but the select-window document says "Select WINDOW which must be a live window".  https://github.com/yuya373/emacs-slack/issues/438#issuecomment-480784846

**Comment by malb** [comment]
Fixed over at  yuya373/emacs-slack#438
